### ISSUE TITLE
IOError: dot exited with status 1

### ISSUE THREAD

**Issue by chris-kuhr** [fpom/snakes#7]
Hi *,

I am new to snakes, trying to get a net rendered on a Linux Mint 17 machine.
If I use Graphviz for drawing directly, it works fine.
But the gv plugin does not.
I have tried both, Python2.7 and Python3.4. I fixed some issues for Python3.4, but it is not working yet.
Here are the standard outputs of both interpreters:

```
~/source_code/python/pyPetriNets $ python2.7 petriNets.py 
Traceback (most recent call last):
  File "petriNets.py", line 148, in <module>
    n.draw(',test-gv')
  File "/usr/local/lib/python2.7/dist-packages/snakes/plugins/gv.py", line 233, in draw
    g.render(filename, engine, debug)
  File "/usr/local/lib/python2.7/dist-packages/snakes/plugins/gv.py", line 142, in render
    raise IOError("%s exited with status %s" % (engine, dot.returncode))
IOError: dot exited with status 1

~/source_code/python/pyPetriNets $ python3.4 petriNets.py 
Traceback (most recent call last):
  File "petriNets.py", line 148, in <module>
    n.draw(',test-gv')
  File "/usr/local/lib/python3.4/dist-packages/snakes/plugins/gv.py", line 233, in draw
    g.render(filename, engine, debug)
  File "/usr/local/lib/python3.4/dist-packages/snakes/plugins/gv.py", line 142, in render
    raise IOError("%s exited with status %s" % (engine, dot.returncode))
OSError: dot exited with status 1
```

Can anyone tell me, what might be wrong, so that I could have a look to fix it?
Or do I need to look into dot?

```
~/source_code/python/pyPetriNets $ dot -v
dot - graphviz version 2.36.0 (20140111.2315)
libdir = "/usr/lib/graphviz"
Activated plugin library: libgvplugin_dot_layout.so.6
Using layout: dot:dot_layout
Activated plugin library: libgvplugin_core.so.6
Using render: dot:core
Using device: dot:dot:core
The plugin configuration file:
        /usr/lib/graphviz/config6
        was successfully loaded.
    render  :  cairo dot fig gd map pic pov ps svg tk vml vrml xdot
    layout  :  circo dot fdp neato nop nop1 nop2 osage patchwork sfdp twopi
    textlayout  :  textlayout
    device  :  canon cmap cmapx cmapx_np dot eps fig gd gd2 gif gv imap imap_np ismap jpe jpeg jpg pdf pic plain plain-ext png pov ps ps2 svg svgz tk vml vmlz vrml wbmp x11 xdot xdot1.2 xdot1.4 xlib
    loadimage   :  (lib) eps gd gd2 gif jpe jpeg jpg png ps svg xbm
```

BR,
Ck

**Comment by fpom** [comment]
Hi,

I run in exactly the same issue from time to time. This is actually dot that crashes on some graphs, I could not understand why. I was hoping this was a regression in Graphviz that some update would fix, but time has passed without such a change. I must take time to report a bug to Graphiv developers.

I'll update this issue when I have news from them.

Franck

**Comment by fpom** [comment]
I've reported the error (issue 0002593) and seen two open bugs complaining about the same crash: "Error: trouble in init_rank".
Hopefully this will be eventually fixed.

**Comment by fpom** [comment]
It looks like the problem is fixed in GraphViz 2.40.1 (which I've just installed from source).
